src/ring_theory/ideal/operations.lean

```diff
@@ -324,15 +324,19 @@ end
 theorem mul_le_inf : I * J ≤ I ⊓ J :=
 mul_le.2 $ λ r hri s hsj, ⟨I.mul_mem_right s hri, J.mul_mem_left r hsj⟩
 
-theorem prod_le_inf {s : finset ι} {f : ι → ideal R} : s.prod f ≤ s.inf f :=
+theorem multiset_prod_le_inf {s : multiset ι} {f : ι → ideal R} :
+  (s.map f).prod ≤ (s.map f).inf :=
```

Comment:
Does this generalize to
```suggestion
  s.prod ≤ s.inf :=
```
Where `s : multiset (ideal R)`? I'd imagine it requires almost no changes.

Reply:
Yep, that works:
```lean
theorem multiset_prod_le_inf (s : multiset (ideal R)) :
  s.prod ≤ s.inf :=
begin
  classical, refine s.induction_on _ _,
  { rw multiset.inf_zero, exact le_top },
  intros a s ih,
  rw [multiset.prod_cons, multiset.inf_cons],
  exact le_trans mul_le_inf (inf_le_inf (le_refl _) ih)
end

theorem prod_le_inf {s : finset ι} {f : ι → ideal R} : s.prod f ≤ s.inf f :=
multiset_prod_le_inf _
```